datalad_metalad/aggregate.py

```diff
@@ -445,10 +445,44 @@ def __call__(
                 into))
 
 
+def _extractor_changed(ds, ds_db, exinfo, path):
+    """Determine whether or not there's a change in the extractors
+    between recorded metadata in `ds_db` and `exinfo`
+
+    Parameters
+    ----------
+
+    ds: Dataset
+      dataset with recorded metadata on `path`
+    ds_db: dict
+      aggregation info of `ds`
+    exinfo: dict
+      extractor configuration to compare to
+    path: Path
+      thing we want metadata about
+    """
+
+    exstate_rec = ds_db.get(path, {}).get('extractors', None)
+
+    # check recorded states in src and dst vs. "current"
+    if (
+            # old aggregate catalog with a plain extractor name list
+            not isinstance(exstate_rec, dict) \
+            or sorted(exinfo.keys()) != sorted(exstate_rec.keys()) \
```

Comment:
(FTR: I'm also looking into #20 and decided to record some notes here)

I don't know the context of

``` py
# old aggregate catalog with a plain extractor name list
            not isinstance(exstate_rec, dict) \
```
but the comment makes it sound like this line checks against a deprecated extractor representation.
If that is true, then by chance, this line also catches if  dataset to be aggregated into does not have any info on the subdataset that is aggregated into it (in which case ``exstate_rec = ds_db.get(path, {}).get('extractors', None)``would lead to ``exstate_rec = None``).

Its desired to catch this, but would it be better to write another explicit statement (instead of catching it (as it seems to me atm) "by chance"?)